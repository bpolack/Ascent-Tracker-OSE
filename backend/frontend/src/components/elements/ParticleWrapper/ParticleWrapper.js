import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

// Import Library Components
import Particles from 'react-particles-js';

export class ParticleWrapper extends Component {

	renderParticles(hidden) {
		if (!hidden) {
			return (
				<Particles className="h-full"
					params={{
						"fps_limit": 30,
						"particles": {
							"collisions": {
								"enable": false
							},
							"number": {
								"value": 6,
								"density": {
									"enable": false
								}
							},
							"size": {
								"value": 260,
								"random": true,
								"anim": {
									"speed": 3,
									"size_min": 0.3
								}
							},
							"opacity": {
								"animation": {
									"speed": 0.33,
									"minimumValue": 0.7
								}
							},
							"line_linked": {
								"enable": false
							},
							"move": {
								"random": true,
								"speed": 0.7,
								"direction": "none",
								"out_mode": "out"
							}
						}
					}} />
			)
		}
	}

	render() {

		const { focus } = this.props;

		return (
			<div className="absolute inset-0">
				{this.renderParticles(focus)}
			</div>
		)
	}
}

ParticleWrapper.propTypes = {
	focus: PropTypes.bool.isRequired
}

const mapStateToProps = state => ({
	focus: state.focus
});

export default connect(mapStateToProps)(ParticleWrapper)