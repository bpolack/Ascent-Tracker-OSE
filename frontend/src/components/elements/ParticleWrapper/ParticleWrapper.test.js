import React from 'react';
import { shallow } from 'enzyme';
import { ParticleWrapper } from './ParticleWrapper';
import Particles from 'react-particles-js';

describe('ParticleWrapper', () => {
	it("Renders correctly without crashing", () => {
		const wrapper = shallow(<ParticleWrapper focus={true} />);
		expect(wrapper).toMatchSnapshot();
	});
	it("Renders the dynamic background if dark mode off", () => {
		const wrapper = shallow(<ParticleWrapper focus={false} />);
		
		const particleElement = wrapper.find(Particles);
		expect(particleElement.exists()).toBeTruthy();
	});
});