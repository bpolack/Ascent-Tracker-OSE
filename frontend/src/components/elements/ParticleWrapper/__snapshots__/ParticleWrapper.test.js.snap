// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticleWrapper Renders correctly without crashing 1`] = `ShallowWrapper {}`;
